packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -1159,22 +1163,25 @@ public void resumePreview() {
         null, (code, message) -> dartMessenger.sendCameraErrorEvent(message));
   }
 
-  public void startPreview() throws CameraAccessException, InterruptedException {
+  public void startPreview(@Nullable Runnable onSuccessCallback)
+      throws CameraAccessException, InterruptedException {
     // If recording is already in progress, the camera is being flipped, so send it through the VideoRenderer to keep the correct orientation.
     if (recordingVideo) {
-      startPreviewWithVideoRendererStream();
+      startPreviewWithVideoRendererStream(onSuccessCallback);
     } else {
-      startRegularPreview();
+      startRegularPreview(onSuccessCallback);
     }
   }
 
-  private void startRegularPreview() throws CameraAccessException {
+  private void startRegularPreview(@Nullable Runnable onSuccessCallback)
+      throws CameraAccessException {
     if (pictureImageReader == null || pictureImageReader.getSurface() == null) return;
```

Comment:
Are the early returns here and below catastrophic situations where never sending the initialization event is the correct behavior? It's at least theoretically a change in behavior from the previous implementation.

Reply:
Good point. I'm not familiar with the state of the camera plugin during initialization. Should that early return code path be treated as an error, throwing on the `camera.initialize()` or should it finish successfully? 
Up until now `camera.initialize()` would not throw, but it seems that executing would mean we are initializing an already closed camera, right?

Reply:
@stuartmorgan-g I've updated those code paths, treating them as no-ops during initialization, like before.